libfreerdp/crypto/certificate.c

```diff
@@ -1281,7 +1297,7 @@ rdpCertificate* freerdp_certificate_new_from_x509(const X509* xcert, const STACK
 
 	if (chain)
 	{
-		cert->chain = sk_X509_dup(chain);
+		cert->chain = sk_X509_deep_copy(chain, X509_dup, X509_free);
```

Comment:
Do we really need to make a deep copy here?
Think `x509_chain_up_ref` should work just fine